dolittle/DotNET.SDK
# Handshake

- Update contracts to latest Gimli prerelease
- Upgrade to .NET6 and using build targets for netcoreapp 3.1, net5 and net6 + step towards .net6-ifying the code
- Introduce the Common project that consists of central building blocks
- Implement a Model-system for centralising declaration of unique ID binding to a "thing" (for Artifacts and Event Processors, ...). This completely revamps the ....Builder classes and the Artifacts project.
- Change ConceptAs to be a record
- Remove Value<> class, as implementations of it should really be 'record' instead
- Implements the contract for performing Handshake with the Runtime
- Add version to DolittleClientConfiguration for setting the head version when performing handshake with the Runtime
- Consolidate SDK API for building EventTypes, AggregateRoots, Event Handlers, ...
- Use interfaces in the callbacks for setup of Dolittle Client in order to hide the Build-methods
- Make the internal building and structure of the DolitleClient cleaner
